### PR TITLE
[primTorch] Added reference for unflatten

### PR DESCRIPTION
**Pull request by fdrocha**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81231
* #81230

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -2523,6 +2524,13 @@ def vstack(tensors: TensorSequenceType) -> TensorLikeType:
     return cat(aligned_tensors, 0)
 
 
+# This not an op, just a method on tensor, so we don't register decomp
```
Comment (Lezcano) [inline torch/_refs/__init__.py:2527]: What do you mean by "we don't register decomp"?

**Review by lezcano — COMMENTED** [review]
This looks very good! Just a small detail.

File: torch/_refs/__init__.py (hunk 1 above)
Comment (fdrocha) [reply torch/_refs/__init__.py:2684]: Please ignore this, I was confused about something. I'll remove this comment.

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -2523,6 +2524,13 @@ def vstack(tensors: TensorSequenceType) -> TensorLikeType:
...
+def unflatten(a: TensorLikeType, dim: int, sizes: ShapeType) -> TensorLikeType:
+    dim = utils.canonicalize_dim(a.ndim, dim)
```
[the remark this answers is not in the document]
Comment (Lezcano) [reply torch/_refs/__init__.py:2688]: We should throw the same errors if possible, so yes, I think it's worth to add a `utils.check` here.

File: test/test_ops.py [hunk 3]
```diff
@@ -1535,6 +1535,7 @@ class TestRefsOpsInfo(TestCase):
         '_refs.broadcast_tensors',
         '_refs.nn.functional.tanhshrink',
         '_refs.swap_axes',
+        '_refs.unflatten',
```
Comment (Lezcano) [inline test/test_ops.py:1538]: Why is this necessary? Now `unflatten` is an ATen op, right?

Comment (fdrocha) [reply test/test_ops.py:1538]: Correct, removed.

**Comment by fdrocha** [comment]
@ngimel could you please take a look?

File: torch/_refs/__init__.py [hunk 4]
```diff
@@ -2675,6 +2676,20 @@ def vstack(tensors: TensorSequenceType) -> TensorLikeType:
     return cat(aligned_tensors, 0)
 
 
+# CompositeImplicitAutograd - don't register decomp
+@out_wrapper()
```
Comment (ngimel) [inline torch/_refs/__init__.py:2680]: I don't think that's correct, `unflatten` always returns a view on the input, and `out_wrapper` will always return a copy. cc @mruberry

Comment (Lezcano) [reply torch/_refs/__init__.py:2680]: I proposed this as an extension of the operation, but it's clearly wrong. I guess no view function has an `out=` variant for this reason, right? We should just remove `out_wrapper` here. Now, given that the tests are passing, we're missing a test testing for this.

Comment (ngimel) [reply torch/_refs/__init__.py:2680]: there was a test for comparing view relationship on out, probably it's skipped/disabled. Yeah, removing `out` wrapper if original op doesn't have `out` sounds fine to me.

**Review by ngimel — APPROVED** [review]
